src/ufuncs.c

```diff
@@ -222,21 +235,30 @@ static void Y_Y_func(char **args, npy_intp *dimensions,
                      npy_intp *steps, void *data)
 {
     FuncGEOS_Y_Y *func = (FuncGEOS_Y_Y *)data;
-    void *context_handle = geos_context[0];
     GEOSGeometry *in1, *ret_ptr;
 
+    GEOS_INIT;
+
     UNARY_LOOP {
         /* get the geometry: return on error */
-        if (!get_geom(*(GeometryObject **)ip1, &in1)) { return; }
+        if (!get_geom(*(GeometryObject **)ip1, &in1)) { errstate = PGERR_NOT_A_GEOMETRY; goto finish; }
         if (in1 == NULL) {
-            /* in case of a missing value: return NULL (NaG) */
+            /* in case of a missing value: return NULL (None) */
             ret_ptr = NULL;
         } else {
-            ret_ptr = func(context_handle, in1);
-            /* trust that GEOS calls HandleGEOSError on error */
+            ret_ptr = func(ctx, in1);
+            // Check last_error if the result is NULL.
+            // We can't be sure otherwise if it is an exception or a 'missing' geometry
```

Comment:
Isn't the case of a missing geometry already handled in the `if (in1 == NULL) {` branch? (so you shouldn't end up here in the `else` when you have a missing geometry?)

Reply:
I delved it a bit and it seems that for some of my own-written functions, NULL can either signify a GEOSException or a missing value. This is the case for:

 - GetPointN, GetInteriorRingN, GetGeometryN when N is out of bounds or the geometry is of incorrect type
 - GetExteriorRing if geometry is not a polygon
 - GEOSBoundaryAllTypes_r when the geometry is a collection

Maybe it is better to set an exception instead in these cases. I'll make a separate issue.